extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/SketchAggregatorFactory.java

```diff
@@ -170,6 +171,25 @@ public int getSize()
     return size;
   }
 
+  @Override
+  public int guessAggregatorHeapFootprint(long rows)
+  {
+    final int maxEntries = size * 2;
+    final int expectedEntries;
+
+    if (rows > maxEntries) {
+      expectedEntries = maxEntries;
+    } else {
+      final int minEntries = 1 << Util.MIN_LG_ARR_LONGS;
```

Comment:
Should `minEntries = 1 << Util.MIN_LG_ARR_LONGS` and the largest possible preamble, `Family.UNION.getMaxPreLongs() << 3` be taken out into constants?